holidays/calendars.py

```diff
@@ -63,6 +63,37 @@ def _islamic_to_gre(g_year: int, h_month: int, h_day: int) -> Iterable[date]:
     return (gre_date for gre_date in gre_dates if gre_date.year == g_year)
 
 
+def _n_weekday_from_date(d: date, weekday: int, n: int = 1) -> date:
+    """
+    Return n-th weekday from date (e.g. 1st Monday, 2nd Saturday).
+    """
+
+    if n > 0:
+        jumpdays = (n - 1) * 7 + (weekday - d.weekday()) % 7
```

Comment:
I'd use `delta_days` or just `days`

Reply:
> I guess it also means that `dateutil` is not very optimized; makes me wonder if `dateutil.easter` can be made faster too (which incidentally would have the side benefit of removing the dependency on `dateutil` altogether... 🙂

Yes, `dateutil.relativedelta` is quite complex: `relativedelta` is a class, `weekday` is a class too. `easter` is a relatively simple function, we can only take it as it is. The most complex thing is `dateutil.parser`…